Cassette/CASQueueFile.m

```diff
@@ -264,41 +319,60 @@ - (void)pop:(NSUInteger)amount {
     for (NSUInteger i = 0; i < amount; i++) {
         totalLengthToErase += ElementHeaderLength + newFirstLength;
         newFirstPosition = [self wrapPosition:newFirstPosition + ElementHeaderLength + newFirstLength];
-        NSData *buffer = [self ringReadAtPosition:newFirstPosition count:ElementHeaderLength];
+        NSData *buffer = [self ringReadAtPosition:newFirstPosition count:ElementHeaderLength error:error];
+        if (!buffer) {
+            return NO;
+        }
         newFirstLength = readUnsignedInt(buffer, 0);
     }
 
     // Commit the header and reassign pertinent in-memory variables
-    [self writeHeader:self.fileLength
-         elementCount:self.elementCount - amount
-        firstPosition:newFirstPosition
-         lastPosition:self.last.position];
+    if (![self writeHeader:self.fileLength
+              elementCount:self.elementCount - amount
+             firstPosition:newFirstPosition
+              lastPosition:self.last.position
+                     error:error]) {
+        return NO;
+    }
     self.elementCount -= amount;
     self.first = [[CASQueueFileElement alloc] initAtPosition:newFirstPosition withLength:newFirstLength];
 
     // Zero out the data where the elements were removed
-    [self ringEraseAtPosition:eraseStartPosition length:totalLengthToErase];
+    return [self ringEraseAtPosition:eraseStartPosition length:totalLengthToErase error:error];
 }
 
 - (void)clear {
-    // Reset the header
-    [self writeHeader:QueueFileInitialLength
-         elementCount:0
-        firstPosition:0
-         lastPosition:0];
-
-    // Zero out the data in our file storage
-    NSData *buffer = [NSMutableData dataWithLength:QueueFileInitialLength - QueueFileHeaderLength];
-    [self ringWriteAtPosition:QueueFileHeaderLength buffer:buffer];
+    [self clearAndReturnError:NULL];
+}
 
+- (BOOL)clearAndReturnError:(NSError * __autoreleasing * _Nullable)error {
     // Reset in-memory variables
     self.elementCount = 0;
     self.first = CASQueueFileElement.null;
     self.last = CASQueueFileElement.null;
+
+    // Reset the header
+    if (![self writeHeader:QueueFileInitialLength
+              elementCount:0
+             firstPosition:0
+              lastPosition:0
+                     error:error]) {
+        return NO;
+    }
+
+    // Zero out the data in our file storage
+    NSData *buffer = [NSMutableData dataWithLength:QueueFileInitialLength - QueueFileHeaderLength];
+    if (![self ringWriteAtPosition:QueueFileHeaderLength buffer:buffer error:error]) {
+        return NO;
+    }
+
```

Comment:
I should have been more clear. "Reset in-memory variables" code block should happen after "Reset the header" code block and before "Zero out the data in our file storage" code block. I'm identifying these code blocks by the comments.

This is to ensure the in-memory representation of these variables stays consistent with the disk representation.

Reply:
Got it! Fixed, thanks.